src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionsUpdateActionUtils.java

```diff
@@ -202,9 +217,27 @@ private static boolean haveSameAttributeType(
         @Nonnull final AttributeType attributeTypeA,
         @Nonnull final AttributeType attributeTypeB) {
 
```

Comment:
I could not understand this change 🤔

Reply:
@ahmetoz This is to support the case of nested type attributes or sets of nested type attributes with different references. Please check the test cases added for it: https://github.com/commercetools/commercetools-sync-java/pull/416/files?file-filters%5B%5D=.java#diff-e76da37897f4fbb30a1b0f712ffe9994

Reply:
what about the enum types?

Reply:
enums are compared after this step at `AttributeDefinitionUpdateActionUtils#buildActions` which means this method has concluded that they are both enums or lenums.